File: src/main/java/com/springsource/greenhouse/recent/RecentActivity.java
package com.springsource.greenhouse.recent;

public class RecentActivity {

	private final String memberPictureUrl;
	
	private final String text;

	private final String imageUrl;

	public RecentActivity(String memberPictureUrl, String text, String imageUrl) {
		this.memberPictureUrl = memberPictureUrl;
		this.text = text;
		this.imageUrl = imageUrl;
	}

	public String getMemberPictureUrl() {
		return memberPictureUrl;
	}

	public String getText() {
		return text;
	}	

	public String getImageUrl() {
		return imageUrl;
	}
	
	public String toString() {
		return "Recent Activity: " + text;
	}
	
}

File: src/main/webapp/resources/recentActivity.js
$(document).ready(function() {
	var basePath = location.protocol + "//" + location.host + greenhouse.contextPath;
	
	function activityNotificationHandler(response) {
		var activityItems = $.parseJSON(response.responseBody);
		for (var i = 0; i < activityItems.length; i++) {
			var activity = activityItems[i];
			$("<li></li>").addClass("newItem").append($("<img/>", { "src" : activity.memberPictureUrl }).addClass("profile")).
				append("<p>" + activity.text + "</p>").
				append($("<img/>", { "src" : activity.imageUrl}).addClass("activity")).
				prependTo("#recentActivityList");	
			if ($("#recentActivityList li").length == 6) {
				$("#recentActivityList li:last").slideUp("slow", function() {
					$(this).remove();
				});
			}
			$("#recentActivityList li.newItem").slideDown("slow", function() {
				$(this).removeClass("newItem");
			});
		}
	}

	/* transport can be : long-polling, streaming or websocket */
	$.atmosphere.subscribe(basePath + '/pubsub/recent/notifications', activityNotificationHandler,
		$.atmosphere.request = {
			transport : "long-polling",
			headers : { "Accept" : "application/json" }
		}
	);
});